WordPress/openverse-api
# Setup Sphinx for developer-docs

## Description


This PR installs and configures Sphinx as a proof of concept to show how documentation for the API's Django codebase would work.

This PR
- adds Sphinx, the Furo theme and an auto-reloading builder
- configures Sphinx for Django support to autogenerate docs for the `catalog` module
- lightly customises the Furo theme for branding with Openverse color palette
- adds `just` recipes for building, hosting and live-reloading the docs site

As a proof-of-concept-ish stage right now, most of the documents in the repo right now, are auto-generated. We can add more long-form documentation `.rst` files later on.

## Testing Instructions

- Clone the repo and check out the `sphinx` branch.
- *Rebuild the `web` image to install dependencies with Pipenv.
- *Ensure that you have a populated `.env` file inside `openverse-api/`.
- Run the live-reload server: `just sphinx-live`. Visit port 3000 on `localhost`.
- Compile the docs: `just sphinx-make`.
- Serve the compiled docs `just sphinx-serve`. Visit port 3000 on `localhost`.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
